src/book/mod.rs

```diff
@@ -270,8 +282,18 @@ impl MDBook {
                     _ => continue,
                 };
 
-                let path = self.source_dir().join(&chapter_path);
-                info!("Testing file: {:?}", path);
+                let cp = match chapter_path.to_str() {
+                    Some(s) => s,
+                    None => "",
+                };
+
+                if chapter_name != "" && ch.name != chapter_name && cp != chapter_name {
+                    info!("Skipping chapter '{}'...", ch.name);
```

Comment:
This all seemed to have reverted back to using an empty string as a sentinel instead of an Option. Is there any particular reason for that?

Reply:
sorry this might be my Rust ignorance, I couldn't figure out how to compare `chapter : Option<&str>` and `ch.name: String`.  Rust complains: "can't compare `std::string::String` with `std::option::Option<&str>`".  So, moving them both to &str solved the problem, but I guess I could instead convert `ch.name` to `Option<&str>` using:
```
let cn: Option<&str> = Some(&ch.name);
```
I pushed this change to see if this is what you prefer...